### DISCUSSION TITLE
Combat screen, cursor shadow is not displayed for hexagons with obstacle

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#6340] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In fheroes2, the cursor shadow is not displayed for the hexagons with obstacle, unlike in the original game.

In original game:

https://user-images.githubusercontent.com/43583994/143688053-86951996-3fed-453e-a258-503ff4f63cee.mp4

In fheroes2:

https://user-images.githubusercontent.com/43583994/143688102-70638d8d-5db4-4be1-b964-9d46958cd2a2.mp4

https://user-images.githubusercontent.com/43583994/143688113-84bbc119-c9a1-456e-816c-37d5145e595e.mp4

### Save file

Here is the test scenario file:
[0 9947.zip](https://github.com/ihhub/fheroes2/files/7611995/0.9947.zip)

### Additional info

Related closed issue: https://github.com/ihhub/fheroes2/issues/4555.

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , showing a shadow for hexes which are unreachable is not user intuitive. Think about this playing the game for the first time. You will surely think that you unit can reach the position, unless you pay big attention to cursor type.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> Hi @LeHerosInconnu , showing a shadow for hexes which are unreachable is not user intuitive. Think about this playing the game for the first time. You will surely think that you unit can reach the position, unless you pay big attention to cursor type.

I agree to leave it the way it currently works in fheroes2. :)
Feel free to close the issue if everybody is okay with that.

**Comment by Branikolog** [comment]
Hello, @ihhub & @LeHerosInconnu .
In my personal point showing a shadow over obstacles could help players to count straight distance to a particular place better. But I don't find it too essential, since flyers, for instance, are able to fly over the whole battle screen, unlike in H3 and the distance for walking troops would be anyway tracked around the obstacles. Unless, we're going to implement "fly" spell, that would allow walking troops go through obstacles. :wink:

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog,

> Unless, we're going to implement "fly" spell, that would allow walking troops go through obstacles. :wink: 

In a way the "Fly" spell already exists, it's the "Teleport" spell. :)

**Comment by zenseii** [comment]
I converted this to a discussion because none of the participants are very convinced about this change.

**Comment by LeHerosInconnu** [comment]
It's okay the way it works currently.
I close the discussion.
